tests/utils/features/query_features.py

```diff
@@ -0,0 +1,448 @@
+
+    :param system_json: a SystemFeatures dictionary
+    :param include_tasks: if not None, will only consolidate tasks present in this list
+    :param exclude_tasks: if not None, will not consolidate tasks present in this list
+    :returns: a dictionary with only feature data
+    '''
+    features = defaultdict(list)
+    for test in system_json['tests']:
+        if include_tasks is not None and TaskIdVariant(test['suite'], test['task_name'], test['variant']) not in include_tasks:
+            continue
+        if exclude_tasks is not None and TaskIdVariant(test['suite'], test['task_name'], test['variant']) in exclude_tasks:
+            continue
+
+        for feature_name in test.keys():
+            if feature_name not in FEATURES:
+                continue
+            for feature in test[feature_name]:
+                if feature not in features[feature_name]:
+                    features[feature_name].append(feature)
+    return features
+
+
+def minus(first: dict[str, list], second: dict[str, list]) -> dict:
+    '''
+    Creates a new dictionary of first - second calculated on values.
+
+    Ex: 
+    first = {"a":["b","c"],"d":["e"]}; second = {"a":["c"], "q":[]}
+
+    first - second == {"a":["b"],"d":["e"]}
+    '''
+    minus = {}
+    for feature, feature_list in first.items():
+        if feature not in second:
+            minus[feature] = feature_list
+        else:
+            m = [item for item in feature_list if item not in second[feature]]
+            if m:
+                minus[feature] = m
+    return minus
+
+
+def list_tasks(system_json: SystemFeatures, remove_failed: bool) -> set[TaskIdVariant]:
+    '''
+    Lists all tasks present in the SystemFeatures dictionary
+
+    :param remove_failed: if true, will only include those tasks where success == True
+    '''
+    tasks = set()
+    for test in system_json['tests']:
+        if not remove_failed or test['success']:
+            tasks.add(
+                TaskIdVariant(test['suite'], test['task_name'], test['variant']))
+    return tasks
+
+
+def diff(retriever: Retriever, timestamp1: str, system1: str, timestamp2: str, system2: str, remove_failed: bool, only_same: bool) -> dict:
+    '''
+    Calculates set(system1_features) - set(system2_features), each at their
+    respective timestamps. 
+
+    :param remove_failed: if true, will remove all instances of tests where success == False
+    :param only_same: if true, will only calculate difference between tests run on both systems
+    '''
+    system1 = retriever.get_single_json(timestamp1, system1)
+    system2 = retriever.get_single_json(timestamp2, system2)
+
+    include_tasks1 = None
+    include_tasks2 = None
+    if remove_failed or only_same:
+        include_tasks1 = list_tasks(system1, remove_failed)
+        include_tasks2 = list_tasks(system2, remove_failed)
+        if only_same:
+            include_tasks2 = include_tasks1 = include_tasks1.intersection(
+                include_tasks2)
```

Comment:
shouldn't this be split?
```suggestion
    if remove_failed:
        include_tasks1 = list_tasks(system1, remove_failed)
        include_tasks2 = list_tasks(system2, remove_failed)

    if only_same:
        include_tasks1 = include_tasks1.intersection(
                include_tasks2)
        include_tasks2 = include_tasks1
```